homeassistant/components/smartthings/__init__.py

```diff
@@ -0,0 +1,201 @@
+"""SmartThings Cloud integration for Home Assistant."""
+
+import asyncio
+import logging
+from typing import Iterable
+
+from aiohttp.client_exceptions import (
+    ClientConnectionError, ClientResponseError)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ACCESS_TOKEN
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.dispatcher import (
+    async_dispatcher_connect, async_dispatcher_send)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+from .config_flow import SmartThingsFlowHandler  # noqa
+from .const import (
+    CONF_APP_ID, CONF_INSTALLED_APP_ID, DATA_BROKERS, DATA_MANAGER, DOMAIN,
+    SIGNAL_SMARTTHINGS_UPDATE, SUPPORTED_PLATFORMS)
+from .smartapp import (
+    setup_smartapp, setup_smartapp_endpoint, validate_installed_app)
+
+REQUIREMENTS = ['pysmartapp==0.3.0', 'pysmartthings==0.4.2']
+DEPENDENCIES = ['webhook']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistantType, config: ConfigType):
+    """Initialize the SmartThings platform."""
+    await setup_smartapp_endpoint(hass)
```

Comment:
Why is this setup during setup? Why not as part of the entry?

Reply:
It's only required once for the whole component.  I originally supported `configuration.yaml`, but removed that to slim down the PR.  Could be moved to entry setup and moved back if I add the `yaml` setup support in the future.

Reply:
Left it where it was because most appropriate.  It's a component-level operation -- not config entry.  See explanation in the other review comment that outlines the data model.